tcbsd-provision-playbook.yaml

```diff
@@ -340,3 +340,32 @@
       when: static_ip_x001_set.changed or dhcp_x001_set.changed
       ansible.builtin.wait_for_connection:
         delay: 2
+
+    # Useful for apps that need PLC access but not Admin-level config change access
+    # We need to manually set the password ourselves later via "doas passwd username"
+    - name: Create or Remove non-admin User
+      ansible.builtin.user:
+        name: "{{ create_username }}"
+        state: "{{ create_user | ternary('present', 'absent') }}"
+        shell: /usr/local/bin/bash'
+
+    # By default, only pubkey and keyboard interactive are enabled
+    # Password access is useful for apps like pmpsdb_client
+    - name: Configure sshd for password access
+      register: sshd_configure
+      ansible.builtin.lineinfile:
+        path: /etc/ssh/sshd_config
+        line: "PasswordAuthentication yes"
+        insertafter: "^#PasswordAuthentication"
```

Comment:
I guess the carrot here in on purpose? @ZLLentz

Reply:
This field is actually a regex, and the ^ indicates the start of a line.
So, ansible reads this as "find a line that starts with `#PasswordAuthentication`, then put `PasswordAuthentication yes` on the next line if it isn't already there."

The context is that the sshd config file looks like this, with all default values included as comments, and I wanted to put this new config line in a reasonable/known place instead of just at the end of the file.
```
# Change to yes to enable built-in password authentication.
# Note that passwords may also be accepted via KbdInteractiveAuthentication.
#PasswordAuthentication no
#PermitEmptyPasswords no

# Change to no to disable PAM authentication
#KbdInteractiveAuthentication yes
```

And then, after ansible runs here:
```
# Change to yes to enable built-in password authentication.
# Note that passwords may also be accepted via KbdInteractiveAuthentication.
#PasswordAuthentication no
PasswordAuthentication yes
#PermitEmptyPasswords no

# Change to no to disable PAM authentication
#KbdInteractiveAuthentication yes
```

When doing this interactively you'd usually uncomment the line and and change "no" to "yes" but doing it like I've done here is simpler for the script and is easier to revert.